### PR TITLE
Error Taxonomy: Add classification for network errors

### PR DESCRIPTION
**Pull request by kwryankrattiger**
Add some classifications for "other" errors from past week

### CODE REVIEW

File: images/upload-gitlab-failure-logs/taxonomy.yaml [hunk 1]
```diff
@@ -2,6 +2,21 @@
 taxonomy:
   version: "0.0.0-dev"
   error_classes:
+    network_timeout:
+      grep_for:
+        - "socket.timeout"
+        - "curl: (28)"
```
Comment (scottwittenburg) [inline images/upload-gitlab-failure-logs/taxonomy.yaml:8]: I think the regexes need testing, I'm fairly certain @mvandenburgh recently fixed some issues with unescaped parens in other regexes, so I think these aren't going to work as-is.

**Review by scottwittenburg — DISMISSED** [review]
Thanks for working on these @kwryankrattiger.  I think testing the regexes will reveal that the parens need to be escaped (at least that, maybe other issues too).  

And if you can include links to example jobs in the description, like in #437, that will help whoever is reviewing, as well as document the kinds of errors the regexes should be catching.

And then this is going to conflict with #440, which I just merged :grimacing: sorry!

File: images/upload-gitlab-failure-logs/taxonomy.yaml (hunk 1 above)
Comment (mvandenburgh) [reply images/upload-gitlab-failure-logs/taxonomy.yaml:8]: Yup, these will need to be escaped. https://github.com/spack/spack-infrastructure/pull/417

File: images/upload-gitlab-failure-logs/taxonomy.yaml [hunk 2]
```diff
@@ -2,6 +2,21 @@
...
+
+    network_error:
+      grep_for:
+        - "curl: (22)"
+        - "HTTP Error 404"
+        - "curl: (60)"
+
+    concretization_error:
+      grep_for:
+        - "does not satisfy"
```
Comment (alalazo) [inline images/upload-gitlab-failure-logs/taxonomy.yaml:18]: I found another string in logs from last week
```suggestion
        - "does not satisfy"
        - "Error: errors occurred during concretization"
```

File: images/upload-gitlab-failure-logs/taxonomy.yaml (hunk 1 above)
Comment (alalazo) [inline images/upload-gitlab-failure-logs/taxonomy.yaml:8]:
```suggestion
        - "socket.timeout"
        - "curl: (28)"
        - "The read operation timed out"
```

File: images/upload-gitlab-failure-logs/taxonomy.yaml [hunk 3]
```diff
@@ -2,6 +2,21 @@
...
+    network_timeout:
+      grep_for:
+        - "socket.timeout"
+        - "curl: (28)"
+
+    network_error:
+      grep_for:
+        - "curl: (22)"
+        - "HTTP Error 404"
+        - "curl: (60)"
```
Comment (alalazo) [inline images/upload-gitlab-failure-logs/taxonomy.yaml:14]:
```suggestion
        - "curl: (22)"
        - "HTTP Error 404"
        - "curl: (60)"
        - "ConnectionResetError:"
```

**Review by alalazo — COMMENTED** [review]
I left a few more pattern to search for, if they seem sensible. Found them while looking at `spack_error` for last week.

File: images/upload-gitlab-failure-logs/taxonomy.yaml (hunk 1 above)
Comment (kwryankrattiger) [reply images/upload-gitlab-failure-logs/taxonomy.yaml:8]: yeah, no testing on this yet. Is the `grep_for` pattern being passed to a regular `grep "my cool pattern"` then?

File: images/upload-gitlab-failure-logs/taxonomy.yaml (hunk 2 above)
Comment (alalazo) [reply images/upload-gitlab-failure-logs/taxonomy.yaml:18]: https://gitlab.spack.io/spack/spack/-/jobs/6342418

File: images/upload-gitlab-failure-logs/taxonomy.yaml (hunk 1 above)
Comment (alalazo) [reply images/upload-gitlab-failure-logs/taxonomy.yaml:8]: https://gitlab.spack.io/spack/spack/-/jobs/6338227/raw

File: images/upload-gitlab-failure-logs/taxonomy.yaml (hunk 3 above)
Comment (alalazo) [reply images/upload-gitlab-failure-logs/taxonomy.yaml:14]: https://gitlab.spack.io/spack/spack/-/jobs/6318173
https://gitlab.spack.io/spack/spack/-/jobs/6316463

**Comment by alalazo** [comment]
@kwryankrattiger Done below each comment.

**Comment by alalazo** [comment]
@kwryankrattiger Not sure you've checked the issue, so in case apologies for double posting, but I also added a few more regex in https://github.com/spack/spack-infrastructure/issues/441

**Comment by kwryankrattiger** [comment]
@mvandenburgh do you want to merge this and update the taxonomy again

**Review by mvandenburgh — APPROVED** [review]

**Comment by mvandenburgh** [comment]
> @mvandenburgh do you want to merge this and update the taxonomy again

Sure, this looks good to me. Looks like @scottwittenburg  just needs to resolve his review

**Review by scottwittenburg — APPROVED** [review]
LGTM, thanks everyone!

**Pull request merged**